### DISCUSSION TITLE
Add Humidity to Weather Forecast

### DISCUSSION THREAD

**Discussion by Daniel-dev22** [home-assistant/architecture#779] [Entity Models] [answered]
Currently ```Humidity``` is only available for current humidity in the weather entity. However, various integration API's support forecast humidity in their default response, home assistant just doesn't do anything with it. It would be great if it can be added given that it is part of a traditional forecast.

https://developers.home-assistant.io/docs/core/entity/weather/

Integrations with forecast humidity support:
```AccuWeather```, ```National Weather Service``` and ```OpenWeatherMap```. (There may be more integrations that support forecast humidity, I only checked these 3.)

[AccuWeather](https://developer.accuweather.com/accuweather-forecast-api/apis/get/forecasts/v1/hourly/1hour/%7BlocationKey%7D) 

[National Weather Service](https://www.weather.gov/documentation/services-web-api#/default/zone_forecast) 

[OpenWeatherMap](https://openweathermap.org/api/one-call-api)

**Comment by gjohansson-ST** [comment]
I would agree to adding it onto the forecast.

**Comment by brenank** [comment]
Noticed this after I created a related PR with this logic (+ adding two other optional attributes): https://github.com/home-assistant/core/pull/76418

**Reply by Daniel-dev22** [reply]
> Noticed this after I created a related PR with this logic (+ adding two other optional attributes): [home-assistant/core#76418](https://github.com/home-assistant/core/pull/76418)

Hopefully this gets traction. It really is surprising for it to not be included already.

**Comment by vingerha** [comment]
I added a new PR for Openweathermap... hope this helps 
https://github.com/home-assistant/core/pull/92106

**Reply by Daniel-dev22** [reply]
> I added a new PR for Openweathermap... hope this helps 
> https://github.com/home-assistant/core/pull/92106

Maybe your pull request can come back now that this architect request has been approved if you're still interested in getting it added?

**Reply by frenck** [reply]
Please note, the weather model changes must go into its own, dedicated, pull request.

**Reply by vingerha** [reply]
I will need to spend a bit of time to rewrite it then, as a part-time-hacking type of person no clue how to split this PR...

**Reply by gjohansson-ST** [reply]
@vingerha let me know if you want to make it and I can support you. Otherwise I can make it.

**Reply by vingerha** [reply]
Thanks....trying to not bother people too much and learning-by-mistake at the same time... it is quite a thing all this dev stuff and outside my regular day2day job (which does support devops...LOL)

**Comment by frenck** [comment, marked as answer]
We have discussed this architectural proposal, and are in agreement that it is weird this property isn't part of the forecast model.

We accept the proposal to add it. Feel free to open up a PR to add it.

../Frenck

**Reply by frenck** [reply]
Added in https://github.com/home-assistant/core/pull/95064
